ruby/ql/lib/codeql/ruby/security/InsecureDownloadCustomizations.qll

```diff
@@ -143,7 +143,7 @@ module InsecureDownload {
       hasUnsafeExtension(req.getAUrlPart().asExpr().getConstantValue().getString())
     }
 
-    override DataFlow::Node getDownloadCall() { result.asExpr().getExpr() = req }
+    override DataFlow::Node getDownloadCall() { result = req }
```

Comment:
If you're worried, we could also override the return type to be `DataFlow::CallNode`?

Reply:
No I was worried the other way around, such that someone makes a subclass of `HTTP::Client::Request::Range` they may not be a `DataFlow::CallNode`... but looking closer at this, it seems like it's just used in the [select of the query](https://github.com/github/codeql/blob/61bf2154cdd6f85ba29df12e2035458d0adf53a6/ruby/ql/src/queries/security/cwe-829/InsecureDownload.ql#L22), so it doesn't really matter.